src/api/core/accounts.rs

```diff
@@ -228,6 +229,32 @@ async fn post_profile(data: JsonUpcase<ProfileData>, headers: Headers, mut conn:
     Ok(Json(user.to_json(&mut conn).await))
 }
 
+#[derive(Deserialize)]
+#[allow(non_snake_case)]
+struct AvatarData {
+    AvatarColor: Option<String>,
+}
+
+#[put("/accounts/avatar", data = "<data>")]
+async fn put_avatar(data: JsonUpcase<AvatarData>, headers: Headers, mut conn: DbConn) -> JsonResult {
+    let data: AvatarData = data.into_inner().data;
+
+    // It looks like it only supports the 6 hex color format.
+    // If you try to add the short value it will not show that color.
+    // Check and force 7 chars, including the #.
+    if let Some(color) = &data.AvatarColor {
+        if color.len() != 7 {
+            err!("The field AvatarColor must be a string with a length of 7 characts")
```

Comment:
There's a typo:

```suggestion
            err!("The field AvatarColor must be a string with a length of 7 characters")
```

However, maybe you should just mention the proper format:

```suggestion
            err!("The field AvatarColor must be a string containing the HEX color in the format #rrggbb")
```

Reply:
Well, it doesn't really matter what we put there. It is never shown to the user.
But ill change the spelling issue at least :)

Reply:
That is true, but it also helps to present the proper info to admins who have access to the log without having them to read the source code. ;-)

Reply:
I have changed it to a better sentence :)